design/proof-erasure-coding.md

```diff
@@ -0,0 +1,419 @@
+Storage proofs & erasure coding
+===============================
+
+Authors: Codex Team
+
+Erasure coding is used for multiple purposes in Codex:
+
+- To restore data when a host drops from the network; other hosts can restore
+  the data that the missing host was storing.
+- To speed up downloads
+- To increase the probability of detecting missing data on a host
+
+For the first two items we'll use a different erasure coding scheme than we do
+for the last. In this document we focus on the last item; an erasure coding
+scheme that makes it easier to detect missing or corrupted data on a host
+through storage proofs.
+
+Storage proofs
+--------------
+
+Our proofs of storage allow a host to prove that they are still in possession of
+the data that they promised to hold. A proof is generated by sampling a number
+of blocks and providing a Merkle proof for those blocks. The Merkle proof is
+generated inside a SNARK to compress it to a small size to allow for
+cost-effective verification on a blockchain.
+
```

Comment:
Merkle proofs inside a SNARK to have an succinct proof is just one of the options. Maybe clarify that you are discussion this option?

Reply:
I meant for this to be a design document, not a research paper. The design that is described here chooses to use a secondary erasure coding scheme with merkle proofs inside a SNARK. As far as I can tell, this design is the only practical one that we've come up with so far?